### DISCUSSION TITLE
Don't know how to disable automatic discovery

### DISCUSSION THREAD

**Discussion by sthibaul** [pypa/setuptools#3611] [Q&A] [answered]
Hello,

I'm trying to fix the build of pocketsphinx-python against version 65 of setuptools: 
https://github.com/sthibaul/pocketsphinx-python/

But I'm getting this:

```
error: Multiple top-level packages discovered in a flat-layout: ['deps', 'swig', 'debian', 'pocketsphinx_disabled'].

To avoid accidental inclusion of unwanted files or directories,
setuptools will not proceed with this build.

If you are trying to create a single distribution with multiple packages
on purpose, you should not rely on automatic discovery.
Instead, consider the following options:

1. set up custom discovery (`find` directive with `include` or `exclude`)
2. use a `src-layout`
3. explicitly set `py_modules` or `packages` with a list of names

To find more information, look for "package discovery" on setuptools docs.
```

I have tried to look at the setuptools documentation, but I fail to find
how to just disable automatic discovery. setup.py really does the job of
listing files etc., so I don't understand what's left to make setuptools
not perform automatic discovery, while previous versions of setuptools
were behaving just fine.

Samuel

**Comment by abravalheri** [comment, marked as answer]
Hi @sthibaul, can you post here a [minimal reproducer](https://stackoverflow.com/help/minimal-reproducible-example) for this error?

In theory, if you specify `packages` in `setup.py` this behaviour should not be triggered. So I need a bit more information to investigate this properly...

**Comment by sthibaul** [comment]
`packages` is specified in `setup.py` yes: https://github.com/sthibaul/pocketsphinx-python/blob/master/setup.py#L285

> can you post here a minimal reproducer

I can give a try, I'm just not sure I'll be able to wade through dependencies between the lines of setup.py.

**Comment by sthibaul** [comment]
> packages is specified in setup.py yes: https://github.com/sthibaul/pocketsphinx-python/blob/master/setup.py#L285

Oops, no, in the particular build system when I encounter the issue, the packages line gets commented. Thanks for confirming which bit needed to be checked again!
